src/menus/menu.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  EmbedBuilder,
+  SelectMenuInteraction,
+  StringSelectMenuBuilder,
+  StringSelectMenuComponent,
+  StringSelectMenuInteraction,
+} from "discord.js";
+import Query from "../routes/Query";
+import { getColor } from "../utilities/profiles/render";
+import Badges from "../utilities/profiles/badges";
+import { ProfileOutput } from "../models/Profile";
+import { CommentOutput } from "../models/Comment";
+
+module.exports = {
+  name: "menu",
+  async execute(
+    interaction: StringSelectMenuInteraction | SelectMenuInteraction
+  ) {
+    await interaction.deferUpdate();
+    let uid = interaction.customId.split("-")[1];
+    let profile = (await Query.profiles.getProfile({ user_id: uid }, 1))
+      .data as ProfileOutput;
+    let badges = Badges.getBadges(profile.badge_flags);
+    switch (interaction.values[0]) {
+      case "badges": {
+        let desc = Badges.getDescriptions(profile.badge_flags).join("\n");
+        if (desc.length == 0) desc = "This user has no badges.";
+        let embed = new EmbedBuilder()
+          .setTitle(`${profile.display_name}'s Badges`)
+          .setColor(getColor(badges.names))
+          .setDescription(`${desc}`);
```

Comment:
Avoid using non-existent abbreviations and abbreviations in general.
```suggestion
        let description = Badges.getDescriptions(profile.badge_flags).join("\n");
        if (description == 0) description = "This user has no badges.";
        let embed = new EmbedBuilder()
          .setTitle(`${profile.display_name}'s Badges`)
          .setColor(getColor(badges.names))
          .setDescription(`${description}`);
```